pkgs/by-name/no/nordvpn/package.nix

```diff
@@ -0,0 +1,200 @@
+  };
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    makeWrapper
+    pkg-config
+  ];
+
+  buildInputs = [
+    # cgo build dependencies go here
+    # https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/go.section.md#envcgo_enabled-var-go-cgo_enabled
+    libxml2'
+  ];
+
+  vendorHash = "sha256-NonkCcSDxLFoFo8XIxyMRO61GbC1fohVHiJH7stnQW8=";
+
+  modPostBuild = ''
+    patch -p0 < ${./gokogiri-xpath-expression.patch}
+  '';
+
+  preBuild = ''
+    # use path $out/bin instead of /usr/lib
+    sed -i "s|/usr/lib/nordvpn|$out/bin|" internal/constants.go
+
+    # use path <<openvpnPatch>>/bin/openvpn instead of /usr/lib/nordvpn/openvpn
+    old_ovpn_path='filepath.Join(internal.AppDataPathStatic, "openvpn")'
+    new_ovpn_path=\"${patchedPkgs.openvpn}/bin/openvpn\"
+    sed -i "s|$old_ovpn_path|$new_ovpn_path|" daemon/vpn/openvpn/config.go
+  '';
+
+  ldflags = [
+    "-X main.Environment=prod"
+    "-X main.Hash=${finalAttrs.version}"
+    # changing the salt would result in nordvpnd crash for existing users
+    "-X main.Salt=e91c1fee12f64ba7a12130a55d5d1d59"
+    "-X main.Version=${finalAttrs.version}"
+  ];
+
+  subPackages = [
+    "cmd/cli"
+    "cmd/daemon"
+    "cmd/norduser"
+  ];
+
+  doCheck = true;
+
+  checkPhase = ''
+    runHook preCheck
+
+    go test ./cli
+    # skip tests that require network access
+    go test ./daemon -skip 'TestTransports|TestH1Transport_RoundTrip'
+    go test ./norduser
+
+    runHook postCheck
+  '';
+
+  postInstall = ''
+    # rename to standard names
+    BIN_DIR=$out/bin
+    install $BIN_DIR/cli $BIN_DIR/nordvpn
+    install $BIN_DIR/daemon $BIN_DIR/nordvpnd
+    install $BIN_DIR/norduser $BIN_DIR/norduserd
+    rm $BIN_DIR/{cli,daemon,norduser}
+
+    # nordvpn needs icons for the system tray and notifications
+    ASSETS_PATH=$out/share/icons/hicolor/scalable/apps
+    install -D assets/icon.svg $ASSETS_PATH/nordvpn.svg
+    for file in assets/*; do
+      install "$file" "$ASSETS_PATH/nordvpn-$(basename $file)"
+    done
+  '';
+
+  postFixup = ''
+    wrapProgram $out/bin/nordvpnd --set PATH ${
+      lib.makeBinPath [
+        e2fsprogs
+        iproute2
+        iptables
+        procps
+        systemdMinimal
+        wireguard-tools
+      ]
+    }
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      categories = [ "Network" ];
+      comment = finalAttrs.meta.description;
+      desktopName = "nordvpn";
+      exec = "nordvpn click %u";
+      icon = "nordvpn";
+      mimeTypes = [ "x-scheme-handler/nordvpn" ];
+      name = "nordvpn";
+      terminal = true;
+      type = "Application";
+    })
+  ];
+
+  meta = {
+    description = "NordVPN application for Linux.";
+    longDescription = ''
+      NordVPN application for Linux.
+      This package currently does not support meshnet.
+      Additionally, if using firewall via `networking.firewall.enable = true;`,
+      then should set `networking.firewall.checkReversePath` to either `"loose"` or `false`.
```

Comment:
I suggest to only mention `"loose"` to not encourage users to completely drop it.